src/Sulu/Component/PHPCR/PathCleanup.php

```diff
@@ -69,30 +81,30 @@ public function cleanup($dirty, $languageCode = null)
                 $dirty = str_replace($key, $value, $dirty);
             }
         }
-
-        $clean = strtolower($dirty);
-
-        // Inspired by ZOOLU
-        // delete problematic characters
-        $clean = str_replace('%2F', '/', urlencode(preg_replace('/([^A-za-z0-9\s\-_\/])/', '', $clean)));
-
         // replace multiple dash with one
-        $clean = preg_replace('/([-]+)/', '-', $clean);
+        $dirty = preg_replace('/([-]+)/', '-', $dirty);
 
         // remove dash before slash
-        $clean = preg_replace('/[-]+\//', '/', $clean);
+        $dirty = preg_replace('/[-]+\//', '/', $dirty);
 
         // remove dash after slash
-        $clean = preg_replace('/\/[-]+/', '/', $clean);
+        $dirty = preg_replace('/\/[-]+/', '/', $dirty);
 
         // delete dash at the beginning or end
-        $clean = preg_replace('/^([-])/', '', $clean);
-        $clean = preg_replace('/([-])$/', '', $clean);
+        $dirty = preg_replace('/^([-])/', '', $dirty);
+        $dirty = preg_replace('/([-])$/', '', $dirty);
+
+        $dirty = str_replace('//', '/', $dirty);
 
-        // remove double slashes
-        $clean = str_replace('//', '/', $clean);
+        $parts = explode('/', $dirty);
+        $newParts = [];
+        foreach ($parts as $part) {
+            $slug = $this->slugger->slug($part, '-', $languageCode);
```

Comment:
looks like this is the reason why the `PageControllerTest::testCopyWithShadow` fails. the previously used regex (`'/([^A-za-z0-9\s\-_\/])/'`) allowed `_` characters. Now they are replaced with `-`. But it would be okay to me to adjust the tests here, i dont think it is good practice to use `_` in urls anymore 🙂